pkgs/games/MarbleMarcher/default.nix

```
@@ -0,0 +1,97 @@
+{ lib, stdenv, fetchFromGitHub, runtimeShell, cmake, eigen, sfml, libglvnd }:
+
+let
+  pname = "MarbleMarcher";
+
+  # The game doesn’t have any versions in GitHub.
+  # Just taking a commit from the top of the “master” branch.
+  version = "git-2019-04-22";
+
+  src = fetchFromGitHub {
+    owner = "HackerPoet";
+    repo = pname;
+    rev = "478afb8ea9c894d6d8b961583aa557e2f63d6198";
+    sha256 = "1kpn9wnf14wm7867dp3lhb10065h4vzy101gab18zkphlwl1dbwq";
+  };
+
+  esc = lib.escapeShellArg;
+
+  assets = stdenv.mkDerivation {
+    pname = "${pname}-assets-pack";
+    inherit version src;
+
+    phases = [ "installPhase" ];
+
+    installPhase = ''
+      set -eu || exit
+      mkdir -p -- "$out"/assets
+      cp -- "$src"/assets/* "$out"/assets
+    '';
+
+    meta = with lib; {
+      homepage = "https://codeparade.itch.io/marblemarcher";
+      description = "Assets for MarbleMarcher videogame";
+      maintainers = with maintainers; [ unclechu ];
+      platforms = platforms.linux;
+
+      # This assets pack contains:
+      # - A couple of fonts (CC-BY-SA 3.0 and SIL OFL 1.1, with license text files)
+      # - Vertex and fragment shaders under GNU/GPLv2 and later
+      # - Images (png, ico) with not clear license
+      # - Audio files (wav, ogg) with not clear license
+      license = licenses.unfree;
+    };
+  };
+
+  unwrappedName = "${pname}-unwrapped";
+
+in stdenv.mkDerivation {
+  inherit pname version src;
+
+  nativeBuildInputs = [ cmake ];
+
+  buildInputs = [ eigen sfml libglvnd ];
+
+  phases = [ "buildPhase" "installPhase" ];
+
+  buildPhase = ''
+    set -eu || exit
+    mkdir build
+    (cd build && cmake "$src")
+    cmake --build build
+
+    ( # Making a wrapper that binds assets with the executable
+      printf '#! %s\n' ${esc runtimeShell}
+      printf 'cd -- %s\n' ${esc (esc assets)}
+      printf 'exec -a "$0" %s/bin/%s "$@"\n' "'$out'" ${esc (esc unwrappedName)}
```

```suggestion
      printf '#! %s\n' ${runtimeShell}
      printf 'cd -- %s\n' ${assets}
      printf 'exec -a "$0" %s/bin/%s "$@"\n' "'$out'" ${unwrappedName}
```
Please use cat with an heredoc instead.

---

I replaced this piece of code with `wrapProgram`.